ros/src/tl_detector/light_classification/tl_detector.py

```diff
@@ -0,0 +1,175 @@
+                output_dict['detection_classes'] = output_dict['detection_classes'][0].astype(np.uint8)
+                output_dict['detection_boxes'] = output_dict['detection_boxes'][0]
+                output_dict['detection_scores'] = output_dict['detection_scores'][0]
+        return output_dict
+
+
+def _load_image_into_numpy_array(image):
+    (im_width, im_height) = image.size
+    return np.array(image.getdata()).reshape(
+            (im_height, im_width, 3)).astype(np.uint8)
+
+
+def _draw_bounding_box_on_image_array(image,
+        ymin,
+        xmin,
+        ymax,
+        xmax,
+        color='red',
+        thickness=4,
+        display_str_list=None):
+    image_pil = Image.fromarray(np.uint8(image)).convert('RGB')
+
+    draw = ImageDraw.Draw(image_pil)
+    im_width, im_height = image_pil.size
+
+    (left, right, top, bottom) = (xmin * im_width, xmax * im_width,
+                                  ymin * im_height, ymax * im_height)
+    draw.line([(left, top), (left, bottom), (right, bottom),
+        (right, top), (left, top)], width=thickness, fill=color)
+    try:
+        font = ImageFont.truetype('arial.ttf', 24)
+    except IOError:
+        font = ImageFont.load_default()
+
+    # If the total height of the display strings added to the top of the bounding
+    # box exceeds the top of the image, stack the strings below the bounding box
+    # instead of above.
+    display_str_heights = [font.getsize(ds)[1] for ds in display_str_list]
+    # Each display_str has a top and bottom margin of 0.05x.
+    total_display_str_height = (1 + 2 * 0.05) * sum(display_str_heights)
+
+    if top > total_display_str_height:
+        text_bottom = top
+    else:
+        text_bottom = bottom + total_display_str_height
+    # Reverse list and print from bottom to top.
+    for display_str in display_str_list[::-1]:
+        text_width, text_height = font.getsize(display_str)
+        margin = np.ceil(0.05 * text_height)
+        draw.rectangle(
+            [(left, text_bottom - text_height - 2 * margin), (left + text_width,
+                                                              text_bottom)],
+            fill=color)
+        draw.text(
+            (left + margin, text_bottom - text_height - margin),
+            display_str,
+            fill='black',
+            font=font)
+        text_bottom -= text_height - 2 * margin
+
+    np.copyto(image, np.array(image_pil))
+
+
+def _visualize_boxes_and_labels_on_image(image,
+        boxes,
+        classes,
+        scores,
+        min_score_thresh=0.25,
+        line_thickness=8):
+    category_index = {
+                1 : 'Green',
+                2 : 'Red',
+                3 : 'Yellow',
+                4 : 'off'
+            }
+    color_index = {
+                1 : 'green',
+                2 : 'red',
+                3 : 'yellow',
+                4 : 'blue'
+            }
+
+    for i in range(20):
```

Comment:
What's the significance of number 20 here?

Reply:
It was default value in [original version](https://github.com/tensorflow/models/blob/master/research/object_detection/utils/visualization_utils.py#L538) for max_boxes_to_draw parameter.